packages/deployments/contracts/contracts/nomad-xapps/contracts/connext/ConnextHandler.sol

```diff
@@ -0,0 +1,524 @@
+   * native asset, routers may use `address(0)` or the wrapped asset
+   * @param _router The router you are adding liquidity on behalf of
+   */
+  function addLiquidityFor(
+    uint256 _amount,
+    address _local,
+    address _router
+  ) external payable override nonReentrant {
+    _addLiquidityForRouter(_amount, _local, _router);
+  }
+
+  /**
+   * @notice This is used by any router to increase their available liquidity for a given asset.
+   * @dev The liquidity will be held in the local asset, which is the representation if you
+   * are *not* on the canonical domain, and the canonical asset otherwise.
+   * @param _amount - The amount of liquidity to add for the router
+   * @param _local - The address of the asset you're adding liquidity for. If adding liquidity of the
+   * native asset, routers may use `address(0)` or the wrapped asset
+   */
+  function addLiquidity(uint256 _amount, address _local) external payable override nonReentrant {
+    _addLiquidityForRouter(_amount, _local, msg.sender);
+  }
+
+  /**
+   * @notice This is used by any router to decrease their available liquidity for a given asset.
+   * @param _amount - The amount of liquidity to remove for the router
+   * @param _local - The address of the asset you're removing liquidity from. If removing liquidity of the
+   * native asset, routers may use `address(0)` or the wrapped asset
+   * @param _to The address that will receive the liquidity being removed
+   */
+  function removeLiquidity(
+    uint256 _amount,
+    address _local,
+    address payable _to
+  ) external override nonReentrant {
+    // transfer to specicfied recipient IF recipient not set
+    address recipient = routerRecipients(msg.sender);
+    recipient = recipient == address(0) ? _to : recipient;
+
+    ConnextUtils.removeLiquidity(_amount, _local, recipient, routerBalances, wrapper);
+  }
+
+  function xcall(XCallArgs calldata _args) external payable override returns (bytes32) {
+    // get remote BridgeRouter address; revert if not found
+    bytes32 remote = _mustHaveRemote(_args.params.destinationDomain);
+
+    ConnextUtils.xCallLibArgs memory libArgs = ConnextUtils.xCallLibArgs({
+      xCallArgs: _args,
+      wrapper: wrapper,
+      nonce: nonce,
+      tokenRegistry: tokenRegistry,
+      domain: domain,
+      home: xAppConnectionManager.home(),
+      remote: remote
+    });
+
+    (bytes32 transferId, uint256 newNonce) = ConnextUtils.xcall(libArgs, adoptedToCanonical, adoptedToLocalPools, relayerFees);
+
+    nonce = newNonce;
+
+    return transferId;
+  }
+
+  /**
+   * @notice Handles an incoming message
+   * @dev This function relies on nomad relayers and should not consume arbitrary amounts of
+   * gas
+   * @param _origin The origin domain
+   * @param _nonce The unique identifier for the message from origin to destination
+   * @param _sender The sender address
+   * @param _message The message
+   */
+  // ROUTER
```

Comment:
nit: Can we get this to be formatted like the other place-markers?

Like:
```
// ============ Nomad Router ============
```
or
```
// ============ Nomad Handler ============
```
whatever yall think

Reply:
deleted